File: structy/array-and-string/pair-product.js
/*
Pair Product
https://structy.net/problems/premium/pair-product
27 August 2023
David Pratt
*/


const pairProduct = (numbers, targetProduct) => {
  let result = [];
  for(let i = 0; i < numbers.length; i++){
    console.log(i);
    for(let j = i + 1; j <= numbers.length; j++){
      console.log(j);
      if(numbers[i] * numbers[j] === targetProduct){
        result.push(i, j)
        break;
      }
    }
  }
  return result;
};

File: structy/array-and-string/pair_product.py
'''
Pair Product
https://structy.net/problems/premium/pair-product
12 November 2023
David Pratt
'''


def pair_product(numbers, target_prod):
  number_count = len(numbers)
  for i in range(number_count):
      for j in range(i + 1, number_count):
          if(numbers[i] * numbers[j] == target_prod):
              return (i, j);
  return False;
  
  

